### PR TITLE
BUG: fix NameError in clip nan propagation tests

### PR DESCRIPTION
**Pull request by Kai-Striega**
The test expects parameters `arr`, `amin` and `amax`. However it then
uses `a`, `amin` and `amax` resulting in a `NameError`. This error was
not caught as the test was marked with xfail.

This commit changes `a` -> `arr` removing the NameError. Further it
adds precautions against similar errors requiring the error to be an
AssertionError and making the test strict.

### CODE REVIEW

File: numpy/core/tests/test_numeric.py [hunk 1]
```diff
@@ -2001,10 +2003,10 @@ def test_clip_problem_cases(self, arr, amin, amax, exp):
     def test_clip_scalar_nan_propagation(self, arr, amin, amax):
         # enforcement of scalar nan propagation for comparisons
         # called through clip()
-        expected = np.minimum(np.maximum(a, amin), amax)
+        expected = np.minimum(np.maximum(arr, amin), amax)
         with assert_warns(DeprecationWarning):
             actual = np.clip(arr, amin, amax)
-            assert_equal(actual, expected)
+        assert_equal(actual, expected)
```
Comment (mattip) [inline numpy/core/tests/test_numeric.py]: Why this change? If possible, only the call that produces the warning should be in the protected block.

Comment (Kai-Striega) [reply numpy/core/tests/test_numeric.py]: That's the change. ``np.clip`` produces the ``DeprecationWarning``. This change moves the assert equal *outside* of the block

Comment (mattip) [reply numpy/core/tests/test_numeric.py]: right, sorry, I read it backwards.

**Comment by eric-wieser** [comment]
I don't really understand the idea behind test in the first  place - the deprecation warning is pretty clear that the current behavior is as expected:
```
DeprecationWarning: Passing `np.nan` to mean no clipping in np.clip has always been unreliable, and is now deprecated. In future, this will always return nan, like it already does when min or max are arrays that contain nan. To skip a bound, pass either None or an np.inf of an appropriate sign.
```

**Comment by Kai-Striega** [comment]
@eric-wieser the problem is that the test does *not* fail due to the warning. It fails due to a NameError in the line ``expected = np.minimum(np.maximum(a, amin), amax)`` as the passed parameter is `arr` not `a`.

**Comment by eric-wieser** [comment]
I understand this patch, and what was broken before. What I don't understand is why we have a test that looks like:
```
with assert_warns_that("nan will not propagate to ret"):
   ret = clip(...)
assert_equal(ret, nan)
```
which is simultaneously assuming a condition `c` then asserting `not c`

**Comment by seberg** [comment]
This is fine, but I agree with Eric, the test was added to be complete (and probably written before the discussion was fully settled). Since I assume we have more tests for the deprecation, I think it would also be fine to just delete it.
@eric-wieser just merge this or just delete the whole thing? (Or else, we just merge it for now, and delete it when the deprecation is gone...)

**Comment by eric-wieser** [comment]
Let's merge it to take the PR count down, could always remove it in a future PR.
